### PR TITLE
[FSDE-13059] Add function to open print dialog with the data table content only

### PR DESCRIPTION
**Pull request by josecarlosgonzalezv**

### CODE REVIEW

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-342/1/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://96dc24224981fc3fcda009bdfe8e95b7e97d9802-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-342-ux-chi.kubeodc-test.corp.intranet/ (internal)

File: src/chi-vue/src/components/data-table/DataTableUtils.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      .map(el => el.outerHTML)
+      .reduce((previousValue, currentValue) => previousValue || '' + currentValue, '');
+
+  win?.document.write(`<head><title>${title}</title>${styleSheets}</head><body class='chi'>${content}</body>`);
+  win?.document.close();
+
+  win?.setTimeout(() => {
+    win.print();
+    win.close();
+  }, 0);
```
Comment (dani-cl-madrid) [inline src/chi-vue/src/components/data-table/DataTableUtils.ts:17]:
```suggestion

  if (!win) {
    throw new Error("Could not open print window. ");
  }
  
  const styleSheets =
    Array.from(document.head.getElementsByTagName('link'))
      .map(el => el.outerHTML)
      .reduce((previousValue, currentValue) => previousValue + currentValue, '') +
    Array.from(document.head.getElementsByTagName('style'))
      .map(el => el.outerHTML)
      .reduce((previousValue, currentValue) => previousValue || '' + currentValue, '');

  win.document.write(`<head><title>${title}</title>${styleSheets}</head><body class='chi'>${content}</body>`);
  win.document.close();

  win.setTimeout(() => {
    win.print();
    win.close();
  }, 0);
```

Comment (dani-cl-madrid) [reply src/chi-vue/src/components/data-table/DataTableUtils.ts:17]: I have just removed the ? after the win object, and checked it is created before working with it. @janibolkvadze, please check my comment and suggest a better option, if you have any. Thanks.

**Review by dani-cl-madrid — CHANGES_REQUESTED** [review]

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://ba63a701c384f68feefa1438aca479145b73f052-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-342-ux-chi.kubeodc-test.corp.intranet/ (internal)

File: src/chi-vue/src/components/data-table/DataTableUtils.ts (hunk 1 above)
Comment (janibolkvadze) [reply src/chi-vue/src/components/data-table/DataTableUtils.ts:17]: Adding checking of `win` and removing the `?` looks good to me.

File: src/chi-vue/src/components/data-table/DataTableUtils.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export function printDataTable(content: string, title: string) {
```
Comment (janibolkvadze) [inline src/chi-vue/src/components/data-table/DataTableUtils.ts:1]: Can we make the function element-agnostic? We might need the util function for some other elements as well in future.
Can we move it to generic `utils` folder and rename it to `printElement`?

**Review by janibolkvadze — CHANGES_REQUESTED** [review]

File: src/chi-vue/src/components/data-table/DataTableUtils.ts (hunk 2 above)
Comment (dani-cl-madrid) [reply src/chi-vue/src/components/data-table/DataTableUtils.ts:1]: +1

Comment (janibolkvadze) [reply src/chi-vue/src/components/data-table/DataTableUtils.ts]: Yes, makes sense. The utility function will be Chi Vue specific.

**Review by janibolkvadze — APPROVED** [review]

**Comment by janibolkvadze** [comment]
@josecarlosgonzalezv Awesome! Great work!

**Review by jllr — APPROVED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-342/3/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://4b7aef75b97c5d18c33dabf8edfdd240dc8ac544-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-342-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Review by dani-cl-madrid — APPROVED** [review]

**Pull request merged**